File: bot.js
const Discord = require('discord.js');
var logger = require('winston');
var auth = require('./auth.json');

// Configure logger settings
logger.remove(logger.transports.Console);
logger.add(logger.transports.Console, {
    colorize: true
});
logger.level = 'debug';
// //Initialize Discord Bot
const bot = new Discord.Client();

bot.on('ready', () => {
    logger.info('Connected');
});

bot.on('message', message => {
	// Bot needs to know if it will execute a command
	// It will listen for messages that will start with `!`
	if (message.content.substring(0, 1) == '!') {
		var args = message.content.substring(1).split(' ');
		var cmd = args[0];

	   
		switch(cmd) {
			case 'help' :
				message.author.send(
					"Here are the commands available to you:\n!help \n!bttpm [message] \n!bttm"
				).catch(console.error);
			break;
			case 'bttpm':
				args.shift();
				var txt = args.join(' ');
				message.channel.send(pokemonText(txt)).catch(console.error);
				message.delete().catch(console.error);
			break;
			case 'bttm' :
				message.channel.fetchMessages({limit: 2})
								.then(messages => {
									if(messages.array()[1].content) {
										messages.array()[1].channel
											.send(pokemonText(messages.array()[1].content)).catch(console.error)};
									messages.array()[0].delete().catch(console.error);
									}
								).catch(console.error);
			break;
			// Just add any case commands if you want to..
		 }
	 }
});

bot.login(auth.token);

function pokemonText(txt) {
	var alt = 0;
	var newTxt = [];
	for (var i = 0, len = txt.length; i < len; i++) {
		if (alt === 0) {
			newTxt[i] = txt[i].toUpperCase();
			alt = 1;
		}
		else {
			newTxt[i] = txt[i].toLowerCase();
			alt = 0;
		}
	}
	return newTxt.join('');
}

File: bot.py
import discord
import logging
import json
import urllib, time

logging.basicConfig(level=logging.INFO)

token = json.load(open('auth.json'))['token']

client = discord.Client()

OPTIONS = {
    'help': 'btthelp',
    'bttpm': 'bttpm',
    'bttm': 'bttm'
}


def pokemon_text(txt: str):
    alt = 0
    new_txt = ""
    for i in range(txt.__len__()):
        if alt == 0:
            new_txt += txt[i].upper()
            alt = 1
        else:
            new_txt += txt[i].lower()
            alt = 0
    return new_txt


@client.event
async def on_ready():
    print("Bot's up. Everything went fine.")
    await client.change_presence(activity=discord.Game(name='!btthelp for list of commands'))


@client.event
async def on_message(message):
    if message.content[0] == '!':
        args = message.content[1:].split(' ')
        cmd = args[0]

        if cmd == OPTIONS['help']:
            await message.author.send(
                f"Here are the commands available to you:\n!{OPTIONS['help']} - display bot usage"
                f"\n!{OPTIONS['bttpm']} [message] - change given message to pokemon text"
                f"\n!{OPTIONS['bttm']} - mock person above")
        elif cmd == OPTIONS['bttpm']:
            await message.channel.send(pokemon_text(' '.join(str(arg) for arg in args[1:])))
            await message.delete()
        elif cmd == OPTIONS['bttm']:
            to_mock = await message.channel.history(limit=2).flatten()
            if to_mock[1]:
                await message.channel.send(pokemon_text(to_mock[1].content))
            await message.delete()
        # Just add any case commands if you want to..


loop_value = True
iteration = 0
while loop_value:
    try:
        urllib.request.urlopen("https://google.com")
        loop_value = False
    except urllib.error.URLError as e:
        print('Network is currently down')
        iteration += 1
        if iteration > 100:
            break
        time.sleep(5)
client.run(token)
